dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -108,6 +106,28 @@
 {
     private final static String TEI_HQL_BY_UIDS = "from TrackedEntityInstance as tei where tei.uid in (:uids)";
 
+    private static final String AND_PSI_STATUS_EQUALS_SINGLE_QUOTE = "and psi.status = '";
+
+    private static final String OFFSET = "OFFSET";
+
+    private static final String LIMIT = "LIMIT";
+
+    private static final String PSI_EXECUTIONDATE = "PSI.executiondate";
+
+    private static final String PSI_DUEDATE = "PSI.duedate";
+
+    private static final String IS_NULL = "IS NULL";
+
+    private static final String IS_NOT_NULL = "IS NOT NULL";
+
+    private static final String SPACE = " ";
+
+    private static final String SINGLE_QUOTE = "'";
+
+    private static final String EQUALS = " = ";
+
+    private static final String PSI_STATUS = "PSI.status";
+
```

Comment:
Suggestion: Maybe here we could move the common keywords in a Utils.

Reply:
Agreed. Discussed this with Lars. As of now, keeping the forward ports similar to 2.36 to begin with. There are lots of possibilities to cleanup and refactor this specific class. I think it needs to be taken as a separate task altogether for 2.37.